docs/docs/guides/atom-effects.md

```diff
@@ -239,6 +241,89 @@ const currentUserIDState = atom({
 });
 ```
 
+## Asynchronous Storage Persistence
+
+If your persisted data needs to be retrieved asynchronously, you can either use a [`Promise`](#promise) in the `setSelf()` function or create an [asynchronous](#asynchronous) function and call that.
+
+### `Promise`
+
+By synchronously calling `setSelf()` with a `Promise` you'll be able to wrap the components inside of the `<RecoilRoot/>` with a `<Suspense/>` component to show a fallback while waiting for `Recoil` to load the persisted values.
+
+The `<Suspense/>` fallback will be shown until the `setSelf()` function has been called.
+
+Note that if the `atoms` later are "reset", they will revert to their default value, and not the initialized value.
+
+```jsx
+const localStorageEffect = key => ({setSelf, onSet}) => {
+  setSelf(
+    localStorage.getItem(key).then((savedValue) => {
+      if (savedValue != null) {
+        return JSON.parse(savedValue);
+      }
+    }),
+  );
+
+  onSet(newValue => {
+    if (newValue instanceof DefaultValue) {
+      localStorage.removeItem(key);
+    } else {
+      localStorage.setItem(key, JSON.stringify(newValue));
+    }
+  });
+};
+
+const currentUserIDState = atom({
+  key: 'CurrentUserID',
+  default: 1,
+  effects_UNSTABLE: [
+    localStorageEffect('current_user'),
+  ]
+});
+```
+
+
+### Asynchronous
+
+Using an asynchronous function your `atoms`, `selectors` etc, will use their default value until the deferred `setSelf()` has been called to set the value.
+
+Unlike using a `Promise`, it is not possible to use `<Suspense/>` to show a fallback.
```

Comment:
"Unlike initializing to a `Promise`, the atom's default value will be used initially, so `<Suspense>` will not show a fallback unless the atom's default is a `Promise` or async selector."

Reply:
Resolved.